src/vm/appdomain.cpp

```diff
@@ -7078,18 +7078,26 @@ EndTry2:;
                     }
                     else if (!fIsWellKnown)
                     {
-                        // Trigger the resolve event also for non-throw situation.
-                        // However, this code path will behave as if the resolve handler has thrown,
-                        // that is, not trigger an MDA.
                         _ASSERTE(fThrowOnFileNotFound == FALSE);
 
-                        AssemblySpec NewSpec(this);
-                        AssemblySpec *pFailedSpec = NULL;
+                        // Don't trigger the resolve event for the CoreLib satellite assembly. A misbehaving resolve event may
+                        // return an assembly that does not match, and this can cause recursive resource lookups during error
+                        // reporting. The CoreLib satellite assembly is loaded from relative locations based on the culture, see
+                        // AssemblySpec::Bind().
+                        if (pSpec->GetCodeBase() != nullptr || !pSpec->IsMscorlibSatellite())
```

Comment:
Oh right, this check is redundant here. It's part of the same check done by AssemblySpec::Bind. The AssemblyResolve event is not raised when a code base is provided anyway, I'll remove the code base check.